helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -73,6 +75,73 @@ public RoutingTableProvider() {
   }
 
   /**
+   * returns the instances for {resource,partition} pair that are in a specific {state} if
+   * aggregateGrouping is turned on, find all resources belongs to the given resourceGroupName and
+   * aggregate all partition states from all these resources.
+   *
+   * @param resourceName
+   * @param partitionName
+   * @param state
+   * @param groupingEnabled
+   *
+   * @return empty list if there is no instance in a given state
+   */
+  public List<InstanceConfig> getInstances(String resourceName, String partitionName, String state,
```

Comment:
Yes, I kept this for back-compatible API.  I will change the new method to getInstancesForResourceGroup,  keep the old getInstances (and mark it as deprecated) and add another method getInstancesForResource.